woocommerce/woocommerce-gateway-stripe
# Add Express Checkouts Customization UI

## Changes proposed in this Pull Request:
Add a menu area for customizing the Express Checkouts buttons(Google Pay / Apple Pay). 

Issue: Link to the GitHub issue this PR addresses (if appropriate).
Fixes #1714 

## Description: 

This was initially scoped to be a separate page using something like `wc_admin_connect_page`, however, if we look at the designs it needs to still "live" under the Payments section area. 

![Screen Shot 2021-09-02 at 12 26 46 PM](https://user-images.githubusercontent.com/6723003/131828039-35a4da34-fe80-4970-8870-18ad0e22204a.png)

Registering a totally independent page removes the top-level Woo settings navigation and feels lost.
Instead, I opted to generate a standalone React entrypoint and conditionally enqueue the app if a specific URL with query args are entered(URL below).

When in the Stripe payments settings area, if the `area` query arg is appended with the value `express_checkouts` the customizer will show.
WCPay uses a `method` query arg to do something similar but it didn't really translate over the same way and as discussed in Slack it was for a different purpose.
I thought instead that `area` was easier to grok and could be re-used in case we needed other submenu React apps that "live" within the Stripe settings area.

## Testing instructions

- Go to `http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&area=express_checkouts` to verify that the Express Checkouts menu loads.
- Double-check that the `Customize Appearance` button link goes to the above URL ^.
- With an Apple Pay and/or Google Pay enabled browser, verify that an Express Checkouts button renders.

## Notes
-------------------
The radio buttons will not affect the button display in this PR since the passing of state between the Customizer menu component and the actual preview button component is done through a set of custom hooks that will live within the /data/ folder. I've opted instead to comment out the use of the hooks and replaced those values with local state.
Thus, you will want to check that state indeed changes when the radio button options are changed.
